### PR TITLE
fix: genomic accession mismatch when getting nearest tx junction 

### LINKED ISSUE #329: "Start chr does not match end chr" when calling `genomic_to_exon_coords` with one position in intronic space
### Describe the bug

When calling `genomic_to_exon_coords` with a start position on an exon and an ending position in intronic space (or vice versa), an older accession, and `nearest_transcript_junction=True`, different procedures (liftover etc) are performed on the two positions. This results in an error about the chromosome accessions of the start and end coordinates being out of joint.

### Steps to reproduce

```python
from cool_seq_tool.app import CoolSeqTool
import asyncio

from cool_seq_tool.schemas import Strand

async def do_thing():
    cst = CoolSeqTool()
    result = await cst.ex_g_coords_mapper.genomic_to_transcript_exon_coordinates(
        alt_ac="NC_000007.13",  # not latest AC for chr 7
        start=73442503,
        end=73457929,  # not on an exon
        strand=Strand.POSITIVE,
        gene="ELN",
        get_nearest_transcript_junction=True
    )
    print(result)

asyncio.run(do_thing())
```

### Expected behavior

n/a

### Current behavior

Gives a warning (really an error):

```
"Start `chr`, NC_000007.14, does not match End `chr`, NC_000007.13"
```

### Possible reason(s)

In the private method `_genomic_to_transcript_exon_coordinate`, there's a logic branch for `get_nearest_transcript_junction`, that ends with an `if` check for whether the position is on an exon.  If it's not, then it's handled there and returned, but if it is, it leaves this section of the code entirely and has additional processing performed in the rest of the method. This appears to result in one accession being swapped in for a newer one. Not sure exactly where the bug is or what the correct response is yet.

```python
if not self._is_exonic_breakpoint(pos, tx_genomic_coords):
    exon = self._get_adjacent_exon(
        tx_exons_genomic_coords=tx_genomic_coords,
        strand=strand,
        start=pos if is_start else None,
        end=pos if not is_start else None,
    )

    params["exon"] = exon
    params["transcript"] = transcript
    params["gene"] = gene
    params["pos"] = pos
    params["chr"] = alt_ac

    self._set_exon_offset(
        params=params,
        start=tx_genomic_coords[exon - 1][3],  # Start exon coordinate
        end=tx_genomic_coords[exon - 1][4],  # End exon coordinate
        pos=pos,
        is_start=is_start,
        strand=strand,
    )
    params["strand"] = strand.value
    resp.transcript_exon_data = TranscriptExonData(**params)
    return resp
# if position is on an exon, doesn't return here, but continues on for the rest of the method
```

### Suggested fix

_No response_

### Branch, commit, and/or version

`main`

### Screenshots

_No response_

### Environment details

n/a

### Additional details

_No response_

### Contribution

Yes, I can create a PR for this fix.

### PR DESCRIPTION
**Pull request by korikuzma**
close #329

* When lifting over to GRCh38 assembly, genomic accession and genomic position should be updated accordingly

### CODE REVIEW

**Review by jarbesfeld — APPROVED** [review]

**Review by jsstevenson — CHANGES_REQUESTED** [review]

File: src/cool_seq_tool/mappers/exon_genomic_coords.py [hunk 1]
```diff
@@ -978,27 +1078,11 @@ async def _get_tx_seg_genomic_metadata(
...
-                return _GenomicTxSeg(
-                    errors=[
-                        f"Position {genomic_pos} does not exist on chromosome {chromosome}"
-                    ]
-                )
-
-            genomic_pos = liftover_data[1]
-            genomic_ac = grch38_ac
+        # Always liftover to GRCh38
+        grch38_data = await self._get_grch38_ac_pos(genomic_ac, genomic_pos)
```
[the remark this answers is not in the document]
Comment (korikuzma) [reply src/cool_seq_tool/mappers/exon_genomic_coords.py:1082]: Ope yes. Thank you. I had added the kwarg for this purpose but forgot to update where it was called!

File: src/cool_seq_tool/mappers/exon_genomic_coords.py [hunk 2]
```diff
@@ -860,6 +907,59 @@ async def _genomic_to_tx_segment(
...
+            chromosome = chromosome[-1].split(":")[-1]
+            liftover_data = self.liftover.get_liftover(
+                chromosome, genomic_pos, Assembly.GRCH38
+            )
+            if liftover_data is None:
+                return _Grch38Data(
+                    accession=None,
+                    position=None,
+                    errors=[
+                        f"Position {genomic_pos} does not exist on chromosome {chromosome}"
```
[the remark this answers is not in the document]
Comment (korikuzma) [reply src/cool_seq_tool/mappers/exon_genomic_coords.py:954]: Updating message to: `f"Lifting over {genomic_pos} on {genomic_ac} from {Assembly.GRCH37.value} to {Assembly.GRCH38.value} was unsuccessful."` . Chromosome is being logged in liftover module

File: src/cool_seq_tool/mappers/exon_genomic_coords.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def _check_errors(
     required_fields: list[str],
     either_or_fields: list[tuple[str, str]] | None = None,
 ) -> dict:
-    """Ensure that required fields are set if `errors` field is empty
+    """Ensure that required fields are set if ``errors`` field is empty
```
Comment (jsstevenson) [inline src/cool_seq_tool/mappers/exon_genomic_coords.py:28]: I also end up using a lot of markdownish/RSTish formatting in docstrings regardless, but fwiw private methods don't show up in the docs so it's NBD if they aren't formatted

File: src/cool_seq_tool/mappers/exon_genomic_coords.py [hunk 4]
```diff
@@ -36,13 +36,15 @@ def _check_errors(
...
+        if not all(
+            values.get(required_field) is not None for required_field in required_fields
+        ):
             err_msg = f"{required_fields} must all be provided"
             raise ValueError(err_msg)
 
         if either_or_fields:
             for field1, field2 in either_or_fields:
-                if not (values.get(field1) or values.get(field2)):
+                if not (values.get(field1) is not None or values.get(field2)):
```
Comment (jsstevenson) [inline src/cool_seq_tool/mappers/exon_genomic_coords.py:47]: @korikuzma could you add parentheses here to ensure the intended logical order is clear? ie 

```python
                if not ((values.get(field1) is not None) or values.get(field2)):
```

assuming I'm reading this right

**Review by jsstevenson — CHANGES_REQUESTED** [review]
one small change request

File: src/cool_seq_tool/mappers/exon_genomic_coords.py (hunk 4 above)
Comment (korikuzma) [reply src/cool_seq_tool/mappers/exon_genomic_coords.py:47]: I messed this one up. I should be checking both

**Comment by korikuzma** [comment]
Made some changes. I reverted back to tuple since it's lightweight. Named tuple / pydantic model seemed a little extra. I can change back if you disagree

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**